daemon/containerd/image_test.go

```diff
@@ -296,3 +298,45 @@ func newTestDB(ctx context.Context, t *testing.T) *metadata.DB {
 
 	return mdb
 }
+
+type testSnapshotterService struct{}
+
+func (s *testSnapshotterService) Stat(ctx context.Context, key string) (snapshots.Info, error) {
+	return snapshots.Info{}, nil
+}
+
+func (s *testSnapshotterService) Update(ctx context.Context, info snapshots.Info, fieldpaths ...string) (snapshots.Info, error) {
+	return snapshots.Info{}, nil
+}
+
+func (s *testSnapshotterService) Usage(ctx context.Context, key string) (snapshots.Usage, error) {
+	return snapshots.Usage{}, nil
+}
+
+func (s *testSnapshotterService) Prepare(ctx context.Context, key, parent string, opt ...snapshots.Opt) ([]mount.Mount, error) {
+	return nil, nil
+}
+
+func (s *testSnapshotterService) View(ctx context.Context, key, parent string, opt ...snapshots.Opt) ([]mount.Mount, error) {
+	return nil, nil
+}
+
+func (s *testSnapshotterService) Commit(ctx context.Context, name, key string, opts ...snapshots.Opt) error {
+	return nil
+}
+
+func (s *testSnapshotterService) Remove(ctx context.Context, key string) error {
+	return nil
+}
+
+func (s *testSnapshotterService) Walk(ctx context.Context, fn snapshots.WalkFunc, filters ...string) error {
+	return nil
+}
+
+func (s *testSnapshotterService) Close() error {
+	return nil
+}
+
+func (s *testSnapshotterService) Mounts(ctx context.Context, key string) ([]mount.Mount, error) {
+	return nil, nil
+}
```

Comment:
It looks like the test only calls 2 methods (`Stat` and `Usage`), so instead of implementing stubs for all methods, you could consider embedding the interface (calling any of the non-implemented ones will panic, so should surface);

```suggestion
type testSnapshotterService struct {
	snapshots.Snapshotter
}

func (s *testSnapshotterService) Stat(ctx context.Context, key string) (snapshots.Info, error) {
	return snapshots.Info{}, nil
}

func (s *testSnapshotterService) Usage(ctx context.Context, key string) (snapshots.Usage, error) {
	return snapshots.Usage{}, nil
}
```

Reply:
I keep forgetting about this "trick", it's super useful though.

Reply:
Yes, thanks, I always forget about it too!